File: 1.Algorithm_Implementations/Sorting/Selection_Sort/Python/selection_sort.py
def selectionSort(array):
    array_length = len(array)
    
    for i in range(array_length):
        index_of_smallest_number = i

        for j in range(i + 1, array_length):
            if array[j] < array[index_of_smallest_number]:
                index_of_smallest_number = j
        swap(i, index_of_smallest_number, array)
        
    return array

# Pythonic way of swapping values
def swap (index1, index2, array):
    array[index1], array[index2] = array[index2], array[index1]

File: 1.Algorithm_Implementations/Sorting/Selection_Sort/Java/SelectionSort.java

class SelectionSort {
    public static int[] sort(int[] array) {
        int smallestNumberIndex;

        for (int i = 0; i < array.length; i++) {
            smallestNumberIndex = i;

            for (int j = i + 1; j < array.length; j++)
                if (array[j] < array[smallestNumberIndex])
                    smallestNumberIndex = j;
            swap(array, i, smallestNumberIndex);
        }
        return array;
    }

    private static void swap(int[] array, int index, int smallestNumberIndex) {
        int tmp = array[index];
        array[index] = array[smallestNumberIndex];
        array[smallestNumberIndex] = tmp;
    }
}
